webapp/src/pages/PlayPage/PlayPage.tsx

```diff
@@ -43,8 +47,17 @@ export default function PlayPage() {
     {
       variant: ButtonVariant.SUBMIT,
       iconVariant: IconVariant.ADD,
-      onClick: () => navigate(PLAY_GAME_URL),
-      children: 'New game',
+      onClick: () => {
+        socket.emit(gameQueueClientToServerWsEvents.gameQueueJoin);
+        setGameCtx &&
+          setGameCtx({
+            isPlaying: false,
+            isWaitingToPlay: true,
+            gameRoomId: null,
+          }); // TODO: these setGameCtx should be broadcasted to all tabs
```

Comment:
tried, couldn't succeed. Also, why?

Reply:
thought: One possible way to do it is to emit a `gameQueueClientToServerWsEvents.gameQueueJoin` event from the client, process the event on the server and emit a `gameQueueServerToClientWsEvents.updateGameContext` to a user (`this.server.to(userId).emit(...)`).

The clients could subscribe to the `updateGameContext` event and update the `gameCtx`.

Reply:
`gameQueueServerToClientWsEvents.updateGameContext`
Your proposition is to create a new event, isn't it? How do I prevent the user that emitted it in the first place from catching this event as well and updating its gameCtx twice? Or you wouldn't update the game context, not even from the client that actually makes the update, until you receive this event from the server? Doesn't this equate to my proposition of adding acknoledgments to every message? Doesn't it make everything a lot more complicated? 
Maybe adding the ack only in this instance doesn't add so much complexity, but we'd have to rehash/revisit all of my diagrams one by one to see that we stay consistent.

Reply:
If we want to broadcast these changes to all the tabs, I think this is one way to do it because each tab runs an webapp instance with a different`socket.id`.

Reply:
I've opened a different issue to tackle this [here](https://github.com/42AGV/ft_transcendence/issues/479).